crates/wasmparser/src/validator/operators.rs

```diff
@@ -196,10 +220,40 @@ impl OperatorValidator {
         Ok(())
     }
 
+    /// Returns the current operands stack height.
     pub fn operand_stack_height(&self) -> usize {
         self.operands.len()
     }
 
+    /// Returns the optional value type of the value operand at the given
+    /// `index` from the top of the operand stack.
+    ///
+    /// Returns `None` if the `index` is out of bounds.
+    ///
+    /// # Note
+    ///
+    /// An `index` of 0 will refer to the last operand on the stack.
+    pub fn peek_operand_at(&self, index: usize) -> Option<Option<ValType>> {
```

Comment:
I really dislike the `Option<Option<ValType>>` return type but also did not want to erase potentially important information by flattening the `Option`. What would you recommend to do here? What cases are there that push a `None` `ValType` btw?

Reply:
I think this is ok for now, this isn't something we need to set in stone from day 1. It's ok to iterate on this over time and improve later. Could the documentation explain the meaning of `Some(None)` though?